sgxs-loaders/src/win_sgx/mod.rs

```diff
@@ -0,0 +1,342 @@
+            ENCLAVE_RETRY => Retry,
+            ENCLAVE_INVALID_SIZE => InvalidSize,
+            ENCLAVE_NOT_INITIALIZED => NotInitialized,
+            _ => Other(error),
+        }
+    }
+}
+#[derive(Fail, Debug)]
+pub enum Error {
+    #[fail(display = "Failed to call CreateEnclave.")]
+    Create(#[cause] LibraryError),
+    #[fail(display = "Failed to call LoadEnclaveData.")]
+    Add(#[cause] LibraryError),
+    #[fail(display = "Failed to call InitializeEnclave.")]
+    Init(#[cause] LibraryError),
+}
+impl EinittokenError for Error {
+    fn is_einittoken_error(&self) -> bool {
+        match self {
+            &Error::Init(LibraryError::InvalidAttribute) |
+            &Error::Init(LibraryError::InvalidMeasurement) |
+            // InvalidEinitToken and InvalidCpusvn get coded this way
+            &Error::Init(LibraryError::Other(ENCLAVE_UNEXPECTED)) => true,
+            _ => false,
+        }
+    }
+}
+
+impl EnclaveLoad for WinInnerLibrary {
+    type Error = Error;
+
+    fn new(
+        device: Arc<WinInnerLibrary>,
+        ecreate: MeasECreate,
+        attributes: Attributes,
+        miscselect: Miscselect,
+    ) -> Result<Mapping<Self>, Self::Error> {
+        let secs = Secs {
+            size: ecreate.size,
+            ssaframesize: ecreate.ssaframesize,
+            miscselect,
+            attributes,
+            ..Default::default()
+        };
+
+        let mut error = 0;
+        let curhandle: HANDLE = unsafe {GetCurrentProcess()};
+        let issupported = unsafe {IsEnclaveTypeSupported(EnclaveType::Sgx as u32)};
+        // set error if issupported is false
+        if issupported == 0 {
+            return Err(Error::Create(LibraryError::NotSupported));
+        }
+        let base = unsafe {
+            CreateEnclave(
+                curhandle,
+                ptr::null_mut(),
+                ecreate.size as _,
+                0,
+                EnclaveType::Sgx as u32,
+                &secs as *const _ as *const c_void,
+                mem::size_of::<Secs>() as u32,
+                &mut error as *mut u32
+            )
+        };
+        if base.is_null() {
+            Err(Error::Create(error.into()))
+        } else {
+            Ok(Mapping {
+                device,
+                tcss: vec![],
+                base: base as _,
+                size: ecreate.size,
+            })
+        }
+    }
+    fn add(
+        mapping: &mut Mapping<Self>,
+        page: (MeasEAdd, PageChunks, [u8; 4096]),
+    ) -> Result<(), Self::Error> {
+
+        let (eadd, chunks, data) = page;
+
+        let mut flags = WinPageProperties::empty();
+        if eadd
+            .secinfo
+            .flags
+            .intersects(SecinfoFlags::PENDING | SecinfoFlags::MODIFIED | SecinfoFlags::PR)
+        {
+            return Err(Error::Add(LibraryError::InvalidParameter));
+        }
+        if eadd.secinfo.flags.intersects(SecinfoFlags::R | SecinfoFlags::W | SecinfoFlags::X) {
```

Comment:
This doesn't do anything. My suggestion was to compute the intersection and then test for equality. Also, I just thought of a way to make sure you have all cases covered:
```rust
match (eadd.secinfo.flags & (SecinfoFlags::R | SecinfoFlags::W | SecinfoFlags::X)) as u8 {
    0b000 => ...,
    0b001 => ...,
    etc.
    0b1000...255 => unreachable!(),
}
```